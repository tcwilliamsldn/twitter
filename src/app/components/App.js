import React from 'react';
import { Route, Switch } from 'react-router-dom';
import ProfilePage from './ProfilePage/ProfilePage';
import StatsPage from './StatsPage/StatsPage';
import NotFoundPage from './NotFoundPage/NotFoundPage';

const App = () => (
    <div>
        <Switch>
            <Route exact path="/:profileName" component={ProfilePage} />
            <Route exact path="/:profileName/stats" component={StatsPage} />
            <Route component={NotFoundPage} />
        </Switch>
    </div>
);

export default App;
