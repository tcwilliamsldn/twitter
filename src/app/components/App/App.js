import React from 'react';
import { Route, Switch } from 'react-router-dom';
import ProfilePage from '../ProfilePage/ProfilePage';
import StatsPage from '../StatsPage/StatsPage';
import Error from '../Error/Error';

const App = () => (
    <main>
        <Switch>
            <Route exact path="/:profileName" component={ProfilePage} />
            <Route exact path="/:profileName/stats" component={StatsPage} />
            <Route render={() => <Error message="404: Page Not Found" />} />
        </Switch>
    </main>
);

export default App;
