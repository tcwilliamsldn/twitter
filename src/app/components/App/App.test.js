import App from './App';
import React from 'react';
import { shallow } from 'enzyme';

describe('App', function() {
    it('renders the application', function() {
        const component = shallow(<App />);
        expect(component).toMatchSnapshot();
    });
});
