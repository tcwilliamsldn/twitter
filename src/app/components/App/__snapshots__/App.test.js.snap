// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the application 1`] = `
<main>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/:profileName"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/:profileName/stats"
    />
    <Route
      render={[Function]}
    />
  </Switch>
</main>
`;
